training/creating_dataset.py

```diff
@@ -0,0 +1,127 @@
+import os.path
+import traceback
+from copy import deepcopy
+from typing import List
+
+import pandas as pd
+
+from training.analysis_functions import (
+    get_merged_bestiaries,
+    unpack_column,
+    DataFrameType,
+)
+
+
+def create_dataframe(
+    books: List[str] = [
+        "../pathfinder_2e_data/pathfinder-bestiary.db",
+        "../pathfinder_2e_data/pathfinder-bestiary-2.db",
+        "../pathfinder_2e_data/pathfinder-bestiary-3.db",
+    ],
+    characteristics: List[str] = [
+        "system/abilities",
+        "system/attributes/ac",
+        "system/attributes/hp",
+    ],
+) -> DataFrameType:
+    """
+    Creates dataframe containing chosen characteristics, level (CR) and source book of monsters from chosen books
+    :param books: list of paths of books to load
+    :param characteristics: list of characteristics to load
+    :return: DataFrame with monsters (NPC) form chosen books and with chosen characteristics and their origin book
+    """
+    loading_methods = {
+        "system/abilities": move_values_level_up(value_name="mod"),
+        "system/attributes/ac": load_subcolumn_as_value("ac"),
+        "system/attributes/hp": load_subcolumn_as_value("hp"),
+    }
+    for i in range(len(books) - 1, -1, -1):
+        if not is_path_correct(books[i]):
+            books.pop(i)
+
+    bestiary = get_merged_bestiaries(books)
+    bestiary = bestiary[bestiary["type"] == "npc"]
+    df = _create_df_with_basic_values(bestiary)
+
+    for characteristic in characteristics:
+        if (loading_method := loading_methods.get(characteristic)) is not None:
+            part = loading_method(get_subcolumn(bestiary, characteristic))
+            df = pd.concat(
+                [
+                    df,
+                    part,
+                ],
+                axis=1,
+            )
+
+    return df
+
+
+def is_path_correct(path: str) -> bool:
+    """
+    Throws exception if file does not exist or has a wrong extension
+    :param path: path to book to load
+    :return: True if path is correct otherwise False
+    """
+    if not os.path.isfile(path):
+        print(f"Path {path} does not exist")
+        return False
+    if not (path.endswith(".db") or path.endswith(".json")):
+        print(f"Expected db or json file, got {path}")
+        return False
+    return True
+
+
+def move_values_level_up(value_name: str):
```

Comment:
Też możemy dać informację co zwraca (przegapiłam to wcześniej w CR), tu by było z tego co rozumiem ` -> Callable[[DataFrameType], DataFrameType]`.
Dla referencji https://docs.python.org/3/library/typing.html#typing.Callable